dashboard/app/models/user.rb

```diff
@@ -889,6 +892,11 @@ def User.track_level_progress_sync(user_id, level_id, script_id, new_result, sub
       user_level.save!
     end
 
+    # Create peer reviews after submitting a peer_reviewable solution
+    if user_level && user_level.submitted && Level.cache_find(level_id).try(:peer_reviewable)
```

Comment:
Why would user_level be nil at this point? If the above block throws, does 895 get executed?

Also you can combine first two into a try statement